### ISSUE TITLE
`session.cookie.domain` not being used correctly

### ISSUE THREAD

**Issue by IgnisDa** [teamhanko/hanko#430]
### Checklist

- [X] I could not find a solution in the existing issues or docs.
- [X] I agree to follow this project's [Code of Conduct](https://github.com/teamhanko/hanko/blob/main/CODE_OF_CONDUCT.md).

### Describe the bug

I am using Hanko for my project [Codefarem](https://github.com/IgnisDa/codefarem). It is live on the subdomain https://codefarem.ignisda.tech. Since it uses a subdomain, I have set `session.cookie.domain` to `.codefarem.ignisda.tech`. However, when the `/finalize` endpoint is called, it sends the domain as `codefarem.ignisda.tech`.
<img width="844" alt="image" src="https://user-images.githubusercontent.com/60938164/205474525-2fa5e862-8c23-4bb0-a6eb-decac4e9b2c6.png">

As a result, the cookie is not set and I immediately get this error: `Your session has expired. Please log in again`.

**TLDR: The leading `.` is not respected.**

### Reproducing the bug

You can try to sign up on https://codefarem.ignisda.tech/auth to see the above problem.

### Logs

```Shell
2022-12-04T04:20:20.990713489Z app[web.1]: 2022/12/04 04:20:20 Using config file: ./config/config.yaml
2022-12-04T04:20:20.996329093Z app[web.1]: 2022/12/04 04:20:20 migrate up
2022-12-04T04:20:21.403762499Z app[web.1]: [POP] 2022/12/04 04:20:21 info - Migrations already up to date, nothing to apply
2022-12-04T04:20:21.404357581Z app[web.1]: [POP] 2022/12/04 04:20:21 info - 0.4063 seconds
2022-12-04T04:20:21.519800089Z app[web.1]: 2022/12/04 04:20:21 Using config file: ./config/config.yaml
2022-12-04T04:20:21.712076042Z app[web.1]: ⇨ http server started on [::]:5000
2022-12-04T04:20:53.978966815Z app[web.1]: {"time":"2022-12-04T04:20:53.977793706Z","time_unix":"1670127653","id":"WPkq3flobQc5g4me7d6HO28FTbUcfTOs","remote_ip":"103.79.169.10","host":"codefarem--authenticator.ignisda.tech","method":"OPTIONS","uri":"/.well-known/config","user_agent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36","status":204,"error":"","latency":7064605,"latency_human":"7.064605ms","bytes_in":0,"bytes_out":0,"referer":"https://codefarem.ignisda.tech/"}
2022-12-04T04:20:54.132266532Z app[web.1]: {"time":"2022-12-04T04:20:54.131768573Z","time_unix":"1670127654","id":"FxX7JLUu7eoz4WHuJaTSBnL2hKUzSIj0","remote_ip":"103.79.169.10","host":"codefarem--authenticator.ignisda.tech","method":"GET","uri":"/.well-known/config","user_agent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36","status":200,"error":"","latency":15354677,"latency_human":"15.354677ms","bytes_in":0,"bytes_out":55,"referer":"https://codefarem.ignisda.tech/"}
2022-12-04T04:20:54.197669256Z app[web.1]: {"time":"2022-12-04T04:20:54.197325709Z","time_unix":"1670127654","id":"4fHFH4EKOUZpeDMNHJn1zb33y0Ffnpll","remote_ip":"103.79.169.10","host":"codefarem--authenticator.ignisda.tech","method":"OPTIONS","uri":"/me","user_agent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36","status":204,"error":"","latency":24101,"latency_human":"24.101µs","bytes_in":0,"bytes_out":0,"referer":"https://codefarem.ignisda.tech/"}
2022-12-04T04:20:54.244449196Z app[web.1]: {"time":"2022-12-04T04:20:54.244288554Z","time_unix":"1670127654","id":"VXF0TC2l9HK6xYO9eu12rsi05IjXl5Kj","remote_ip":"103.79.169.10","host":"codefarem--authenticator.ignisda.tech","method":"GET","uri":"/me","user_agent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36","status":400,"error":"code=400, message=missing or malformed jwt","latency":3280544,"latency_human":"3.280544ms","bytes_in":0,"bytes_out":50,"referer":"https://codefarem.ignisda.tech/"}
2022-12-04T04:21:02.622613532Z app[web.1]: {"time":"2022-12-04T04:21:02.622447566Z","time_unix":"1670127662","id":"gFqkL8CgCDIJ70Ch7x1V9hjTC3RrJCKP","remote_ip":"103.79.169.10","host":"codefarem--authenticator.ignisda.tech","method":"OPTIONS","uri":"/user","user_agent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36","status":204,"error":"","latency":41210,"latency_human":"41.21µs","bytes_in":0,"bytes_out":0,"referer":"https://codefarem.ignisda.tech/"}
2022-12-04T04:21:02.785348796Z app[web.1]: {"time":"2022-12-04T04:21:02.785177201Z","time_unix":"1670127662","id":"iecjyyV8Z4QSll149jagCq5eVOwEDeCl","remote_ip":"103.79.169.10","host":"codefarem--authenticator.ignisda.tech","method":"POST","uri":"/user","user_agent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36","status":200,"error":"","latency":100436778,"latency_human":"100.436778ms","bytes_in":39,"bytes_out":94,"referer":"https://codefarem.ignisda.tech/"}
2022-12-04T04:21:02.906922710Z app[web.1]: {"time":"2022-12-04T04:21:02.906510593Z","time_unix":"1670127662","id":"eqJVw9PhS4hw2IxYzvVoKx5andQG3E3d","remote_ip":"103.79.169.10","host":"codefarem--authenticator.ignisda.tech","method":"OPTIONS","uri":"/passcode/login/initialize","user_agent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36","status":204,"error":"","latency":33630,"latency_human":"33.63µs","bytes_in":0,"bytes_out":0,"referer":"https://codefarem.ignisda.tech/"}
2022-12-04T04:21:06.081752558Z app[web.1]: {"time":"2022-12-04T04:21:06.081568289Z","time_unix":"1670127666","id":"DFYs1YVmRLLEbS9DdKzWwz6VAQiFOosW","remote_ip":"103.79.169.10","host":"codefarem--authenticator.ignisda.tech","method":"POST","uri":"/passcode/login/initialize","user_agent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36","status":200,"error":"","latency":3116827230,"latency_human":"3.11682723s","bytes_in":50,"bytes_out":102,"referer":"https://codefarem.ignisda.tech/"}
2022-12-04T04:21:12.784945353Z app[web.1]: {"time":"2022-12-04T04:21:12.784781264Z","time_unix":"1670127672","id":"XohMZdNWWHvMYfqe1G66ygidYi0fOezA","remote_ip":"103.79.169.10","host":"codefarem--authenticator.ignisda.tech","method":"OPTIONS","uri":"/passcode/login/finalize","user_agent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36","status":204,"error":"","latency":66501,"latency_human":"66.501µs","bytes_in":0,"bytes_out":0,"referer":"https://codefarem.ignisda.tech/"}
2022-12-04T04:21:13.131327206Z app[web.1]: {"time":"2022-12-04T04:21:13.131192784Z","time_unix":"1670127673","id":"qxQ4MAsrU52GTlVKpLOiOjzJO8AS3ItL","remote_ip":"103.79.169.10","host":"codefarem--authenticator.ignisda.tech","method":"POST","uri":"/passcode/login/finalize","user_agent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36","status":200,"error":"","latency":303755999,"latency_human":"303.755999ms","bytes_in":61,"bytes_out":99,"referer":"https://codefarem.ignisda.tech/"}
2022-12-04T04:21:13.208364260Z app[web.1]: {"time":"2022-12-04T04:21:13.208209787Z","time_unix":"1670127673","id":"46gg7F8N0mtRkkUpgPdviYKB44UhxRv3","remote_ip":"103.79.169.10","host":"codefarem--authenticator.ignisda.tech","method":"OPTIONS","uri":"/me","user_agent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36","status":204,"error":"","latency":43679,"latency_human":"43.679µs","bytes_in":0,"bytes_out":0,"referer":"https://codefarem.ignisda.tech/"}
2022-12-04T04:21:13.353965933Z app[web.1]: {"time":"2022-12-04T04:21:13.353808251Z","time_unix":"1670127673","id":"tjVhAxFrSm2lCHjQCUvTVPfWy3c8oIUm","remote_ip":"103.79.169.10","host":"codefarem--authenticator.ignisda.tech","method":"GET","uri":"/me","user_agent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36","status":400,"error":"code=400, message=missing or malformed jwt","latency":952281,"latency_human":"952.281µs","bytes_in":0,"bytes_out":50,"referer":"https://codefarem.ignisda.tech/"}
```

### Configuration

```yml
database:
  dialect: postgres
  user: '<redacted>'
  password: '<redacted>'
  host: '<redacted>'
  port: <redacted>
  database: '<redacted>'

passcode:
  email:
    from_address: '<redacted>'
    from_name: 'Codefarem'
  smtp:
    host: '<redacted>'
    port: <redacted>
    user: '<redacted>'
    password: '<redacted>'

secrets:
  keys:
    - '<redacted>'

service:
  name: 'Codefarem'

server:
  public:
    address: ':5000'
    cors:
      enabled: true
      allow_credentials: true
      allow_origins:
        - '*'

session:
  lifespan: '24h'
  cookie:
    domain: '.codefarem.ignisda.tech'
    http_only: true
    secure: true
```

### Hanko Version

v0.3.1

### OS

Linux

### OS Version

`Linux main 5.4.0-131-generic #147-Ubuntu SMP Fri Oct 14 17:07:22 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux`

### Browser Version

Happens on all browsers

### Environment

Docker

### Additional Context

I looked into how https://example.hanko.io works:

<img width="608" alt="image" src="https://user-images.githubusercontent.com/60938164/205474721-96ec136a-cd7b-4127-a2bd-8c93b3ad189d.png">

It sends the `Domain` attribute the same way yet it works fine. Is there something I am doing wrong?

**Comment by IgnisDa** [comment]
It worked when I moved the hanko domain from `codefarem--authenticator.ignisda.tech` to `authenticator.codefarem.ignisda.tech`.

**Comment by FlxMgdnz** [comment]
Hey, I'm glad you got it working. If you have any recommendations where we can improve our docs / readme please let us know.

**Comment by IgnisDa** [comment]
@FlxMgdnz I tried to use the hanko cloud instance, and see that the `set-cookie` header does not include the `Domain` attribute, but on my self hosted instance, the `Domain` is always set.

I use this config:

```yml
session:
  lifespan: '${SESSION_LIFESPAN:-24h}'
  cookie:
    domain: '' # i have also tried to remove this completely
    http_only: true
    secure: true
    same_site: 'strict'
  enable_auth_token_header: true
```

**Comment by spwoodcock** [comment]
Using settings

```yaml
  cookie:
    name: "mydomain_sso"
    domain: ".mydomain.org"
    retention: persistent
    secure: true
    http_only: true
    same_site: "lax"
```

I correctly get a cookie set to `domain=.mydomain.org`, so I think everything is working as intended 👍 
(this can probably be closed)

**Comment by FlxMgdnz** [comment]
> Using settings
> 
>   cookie:
>     name: "mydomain_sso"
>     domain: ".mydomain.org"
>     retention: persistent
>     secure: true
>     http_only: true
>     same_site: "lax"
> I correctly get a cookie set to `domain=.mydomain.org`, so I think everything is working as intended 👍 (this can probably be closed)

Thanks!
